src/workerd/jsg/resource.h

```diff
@@ -191,6 +194,254 @@ template <typename TypeWrapper,
     typename Indexes>
 struct MethodCallback;
 
+// Implements V8 Fast API callback for a method. This class handles both the fast and slow paths.
+template <typename TypeWrapper,
+    const char* methodName,
+    bool isContext,
+    typename T,
+    typename Method,
+    Method method,
+    typename Indexes>
+struct FastMethodCallback;
+
+// Specialization for instance methods compatible with Fast API
+template <typename TypeWrapper,
+    const char* methodName,
+    bool isContext,
+    typename T,
+    typename U,
+    typename Ret,
+    typename... Args,
+    Ret (U::*method)(Args...),
+    size_t... indexes>
+struct FastMethodCallback<TypeWrapper,
+    methodName,
+    isContext,
+    T,
+    Ret (U::*)(Args...),
+    method,
+    kj::_::Indexes<indexes...>> {
+
+  template <bool trackCallCount = false>
+  static void slowCallback(const v8::FunctionCallbackInfo<v8::Value>& args) {
```

Comment:
It seems like we could merge `FastMethodCallback` into `MethodCallback`. They have the same template parameters. `slowCallback()` just calls `MethodCallback::callback`, instead we could just drop it. `MethodCallback` would have `callback` and `fastCallback` methods. This would eliminate a lot of boilerplate?

Reply:
Yeah you're right. Updating now.